save-core/src/commonMain/kotlin/org/cqfn/save/core/Save.kt

```diff
@@ -27,20 +49,79 @@ class Save(
 
     /**
      * Main entrypoint for SAVE framework. Discovers plugins and calls their execution.
+     *
+     * @throws PluginException when we receive invalid type of fixPluginConfig
      */
     fun performAnalysis() {
+        // FixMe: now we work only with the save.toml config and it's hierarchy, but we should work properly here with directories as well
+        // constructing the file path to the configuration file
+        val fullPathToConfig = saveProperties.testRootPath!!.toPath() / saveProperties.testConfigName!!.toPath()
         // get all toml configs in file system
-        val testConfig = ConfigDetector().configFromFile(saveProperties.testConfig)
+        val testConfig = ConfigDetector().configFromFile(fullPathToConfig)
+
+        val plugins: List<Plugin> = discoverPluginsAndUpdateTestConfig(testConfig)
+            // FixMe: Kirill should add merging of configs here
+            // ignoring the section with general configuration as it is not needed to create plugin processors
+            .pluginConfigs.filterNot { it is GeneralConfig }
+            .map {
+                when (it) {
+                    is FixPluginConfig -> FixPlugin(testConfig)
+                    is WarnPluginConfig -> WarnPlugin(testConfig)
+                    else -> throw PluginException("Unknown type <${it::class}> of plugin config was provided")
+                }
+            }
 
-        val plugins: List<Plugin> = emptyList()  // todo: discover plugins (from configuration blocks in TestSuiteConfig?)
         logInfo("Discovered plugins: $plugins")
         plugins.forEach { plugin ->
             logInfo("Execute plugin: ${plugin::class.simpleName}")
-            plugin.execute(testConfig).forEach(this::handleResult)
+            plugin.execute().forEach(this::handleResult)
             logInfo("${plugin::class.simpleName} successfully executed!")
         }
     }
 
+    private fun discoverPluginsAndUpdateTestConfig(testConfig: TestConfig): TestConfig {
+        val testConfigPath = testConfig.location.toString()
+        val parsedTomlConfig = TomlParser(testConfigPath).readAndParseFile()
+        parsedTomlConfig.getRealTomlTables().forEach { tomlPluginSection ->
+            // adding a fake file node to restore the structure and parse only the part of te toml
+            val fakeFileNode = TomlFile()
+            tomlPluginSection.children.forEach {
+                fakeFileNode.appendChild(tomlPluginSection)
```

Comment:
Why do we iterate over `tomlPluginSection.children()` and add `tomlPluginSection` instead of children?
```suggestion
                fakeFileNode.appendChild(it)
```

Reply:
Oh, it's a very serious bug. Don't know why it didn't strike